bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/AudioManager.java

```diff
@@ -0,0 +1,460 @@
+        File file = new File(System.getProperty(ConfigConstants.CONFIG_DIR_PROG_ARGUMENT) + File.separator + SOUND_DIR
+                + File.separator + fileName);
+        FileAudioStream is = new FileAudioStream(file);
+        play(is, sink);
+    }
+
+    /**
+     * Stream audio from the passed url using the default audio sink.
+     *
+     * @throws AudioException in case the url stream cannot be opened
+     */
+    public void stream(String url) throws AudioException {
+        stream(url, null);
+    }
+
+    /**
+     * Stream audio from the passed url to the given sink
+     *
+     * @param url The url to stream from
+     * @param sinkId The id of the audio sink to use or null
+     * @throws AudioException in case the url stream cannot be opened
+     */
+    public void stream(String url, String sinkId) throws AudioException {
+
+        AudioStream audioStream = new URLAudioStream(url);
+
+        AudioSink sink = null;
+        if (sinkId == null) {
+            sink = getSink();
+        } else {
+            sink = audioSinks.get(sinkId);
+        }
+
+        if (sink != null) {
+            try {
+                sink.process(audioStream);
+            } catch (UnsupportedAudioFormatException e) {
+                logger.error("Error playing '{}': {}", url, e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Retrieves the current volume of a sink
+     *
+     * @param sinkId the sink to get the volume for
+     * @return the volume as a value between 0 and 1
+     */
+    public float getVolume(String sinkId) {
+        AudioSink sink = null;
+        if (sinkId == null) {
+            sink = getSink();
+        } else {
+            sink = audioSinks.get(sinkId);
+        }
+
+        if (sink != null) {
+            try {
+                return sink.getVolume();
+            } catch (IOException e) {
+                logger.error("An exception occured while getting the volume of sink {} : '{}'", sink.getId(),
+                        e.getMessage());
+            }
+        }
+
+        return 0;
+
+    }
+
+    /**
+     * Sets the volume for a sink.
+     *
+     * @param sinkId the sink to set the volume
+     * @param volume the volume to set as a value between 0 and 1
+     */
+    public void setVolume(String sinkId, float volume) {
+        AudioSink sink = null;
+        if (sinkId == null) {
+            sink = getSink();
+        } else {
+            sink = audioSinks.get(sinkId);
+        }
+
+        if (sink != null) {
+            try {
+                sink.setVolume(volume);
+            } catch (IOException e) {
+                logger.error("An exception occured while setting the volume of sink {} : '{}'", sink.getId(),
+                        e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Retrieves an AudioSource.
+     * If a default name is configured and the service available, this is returned. Otherwise, the first available
+     * service is returned.
```

Comment:
This does not fit to the description of the return value and the implementation.
If a default name is configured, but the service is not available, null is returned and not the first available service.